comms/src/peer_manager/node_id.rs

```diff
@@ -104,6 +105,19 @@ impl TryFrom<&[u8]> for XorDistance {
     }
 }
 
+impl TryFrom<XorDistance> for u128 {
+    type Error = String;
+
+    fn try_from(value: XorDistance) -> Result<Self, Self::Error> {
+        if value.0.len() > 16 {
+            return Err("XorDistance has too many bytes to be converted to U128".to_string());
+        }
+        let mut bytes: [u8; 16] = [0u8; 16];
+        bytes.iter_mut().set_from(value.0.iter().cloned());
+        Ok(u128::from_be_bytes(bytes))
```

Comment:
If you use this technique, you can remove the `itertools` dependency

```suggestion
         let slice = value.as_bytes();
         if value.0.len() > 16 {
            return Err("XorDistance has too many bytes to be converted to U128".to_string());
        }
        let mut bytes: [u8; 16] = [0u8; 16];
        bytes[..XorDistance::byte_length()].copy_from_slice(&slice[..XorDistance::byte_length()]);
        Ok(u128::from_be_bytes(bytes))
```

```rust
    /// Returns the number of bytes required to represent the `XorDistance`
    pub const fn byte_length() -> usize {
        NODE_XOR_DISTANCE_ARRAY_SIZE
    }
```

Reply:
But I love itertools! alright done